### ISSUE TITLE
Pronounceable password generator (not passphrase)

### ISSUE THREAD

**Issue by Weej1** [keepassxreboot/keepassxc#3784]
[TIP]:  # ( Provide a general summary of the feature in the title above ^^ )
[TIP]:  # ( DO NOT include screenshots of your actual database! )

## Summary
Keepass has a plugin that creates pronounceable passwords (even if they make no sense). These dont use actual words (like a passphrase), but rather grammatical rules to create human pronounceable words.  

## Desired Behavior
KPXC's random password generator is great, ideally a new  separate tab for Pronounceable passwords could be added.  

## Possible Solution
My only reference is the plugin I was using previously for KeePass (not XC) which can be found here: https://sourceforge.net/projects/pronouncepwgen/files/pronouncepwgen/1.0.4b/ 
It has a list of possible dictionaries and certain rules to make a password pronounceable (in English).

### Why does it matter to me: 
 * I find these easier to remember in case KeepassXC cannot be run (reason below)

 *  The words have this effect where you end up making  associations (sometimes!) e.g. N64Botahkac@RajWurl! When I first saw that I immediately broke it down (mentally and visually) to "N64 (as in Nintendo 64), Butter, Cake, at, Raj, World, ! ) " I haven't used this password but the fact that I remember it 8 months later is why I think this tool could be useful.

 * passphrases, while a great alternative can sometimes get too wordy just to get equivalent entropy (e.g. a 16 character password of ~ 80bit entropy will usually generate a 7 word passphrase with each word having 7 ~ 9 characters (excluding separator) to get equal entropy - compare that to above example with equal entropy.        

### What unique circumstances i've had: 
* usually due to location and time constraints e.g. another laptop where I cannot access the internet to download KPXC or when the android app I use is just not feasible to show on a projector screen during a presentation.

* I've also customised pronounceable passwords to suit me better while still looking like a random password.

**Comment by droidmonkey** [comment]
We will not be implementing this.

**Comment by droidmonkey** [comment]
It's literally the antithesis of our program's goals. We don't support lowering your password entropy. Anything done to force patterns in your password makes it dramatically less secure than a random password. If you really want these types of passwords, make them on a website that offers this capability.

**Comment by tasuki** [comment]
I'd also like this, though understand why you don't.

For those who like pronounceable passwords, here's a bash script to create them:

```
#!/bin/bash

if [[ -z "$1" ]]; then
	LENGTH=16
else
	LENGTH=$1
fi

VOWELS=(a e i o u y)
CONSONANTS=(b c d f g h j k l m n p q r s t v w x z)
PASSWORD=""

for ((i=0; i < $LENGTH / 2; i++)); do
	PASSWORD+="${CONSONANTS[$RANDOM % ${#CONSONANTS[@]}]}"
	PASSWORD+="${VOWELS[$RANDOM % ${#VOWELS[@]}]}"
done

echo "$PASSWORD"
```

**Comment by robsdedude** [comment]
> We don't support lowering your password entropy

I'm sad to see such a rigorous stance against this. I also would very much like such a feature. One can still make the password longer to increase the entropy again. After all keepassxc *does* support passphrases using a word list which offers less entropy (at the same length) than a random pronounceable password.

I much prefer pronounceable passwords over random passwords and passphrases as they strike a good balance (given a target entropy) between shortness and easiness to type (for those instances where I have to copy the passwrod manually).

**Comment by s7eph4n** [comment]
> It's literally the antithesis of our program's goals. We don't support lowering your password entropy. Anything done to force patterns in your password makes it dramatically less secure than a random password. If you really want these types of passwords, make them on a website that offers this capability.

Typical case of developer arrogance, treating users as stupid children: "I can't imagine a valid use case, therefore there is no valid use case."

In terms of security the proposed workaround is way worse than the feature requested by OP.

**Comment by droidmonkey** [comment]
No, it's called security by design, also known as fail closed. If you are making pronouncable passwords, then you have no use for a password _manager_. The very act of wanting something pronouncable is so you can remember it. The whole point of KeePassXC is to eliminate the human aspect of passwords.

**Comment by robsdedude** [comment]
> Typical case of developer arrogance

@s7eph4n  Really no need to get personal.


@droidmonkey I'm unsure you considered my arguments in https://github.com/keepassxreboot/keepassxc/issues/3784#issuecomment-1525077691 though.

> It's called security by design.

Why does KeePassXC support word lists/passphrases then? Depending on the dictionary (assuming a standard English one), they provide a lower entropy per characters ratio than pronounceable passwords.

> The whole point of KeePassXC is to eliminate the human aspect of passwords.

It's valid if this is your design goal for KeePassXC in particular. For password managers in general, I disagree. There's still a human being using the manager and some integrations will still require human interaction (e.g., having to copy the password yourself because you're on somebody else's device, or on a device you're newly setting up, or ...).

**Comment by droidmonkey** [comment]
Passphrases are one concession we made, but we have been consistent about not introducing additional parameters or tweaks to them. Passphrases are arguably more secure than pronouncable passwords by sheer size alone. If an attacker doesn't know you are using a passphrase then that in and of itself is added entropy.

However, every single brute force attack scenario will certainly do these two things:

1. Start with an english dictionary
2. Create iterations on words using known replacement values

Those two things are exactly pronouncable passwords.

**Comment by s7eph4n** [comment]
> @s7eph4n Really no need to get personal.

Yes, and I apologize.

Still, the argument doesn't make sense. We are talking about a password _manager_. Password management is more than just filling in passwords. There is plenty of use cases where I want to store my passwords in a password manager but still fill them in by hand. And as pointed out, following @droidmonkey's reasoning passphrases don't make sense (so they should probably be removed). Moreover, it is easily possible to give users guidance on the security of a pronounceable password (e.g. using entropy based on syllables).

If you don't want to implement the feature, fine, it's your app, your project, your workload on implementation and maintenance. But please don't tell users what's good for them.

**Comment by droidmonkey** [comment]
You won't catch me being a fan of passphrases. That was probably the 5th feature we merged after forking KeePassXC. If it was proposed today I'd reject it as well.

**Comment by mlavi** [comment]
I adopted KeePassXC a few years ago to get my accounts under control. I've worked to progressively update my passwords to passphrases for higher entropy and fallback for manual entry. But I've manually entered a password a handful of times over the years (only on mobile, before adding a KeePass database compatible app). This thread has convinced me I don't need passphrases, I'll go fully random. Thanks for everyone offering their perspective!

**Comment by HoroTW** [comment]
For anyone who really wants something like that, you can emulate such a thing using a word list.
I share my created one which can be used in instances where I'm not able to use my pw manager to directly input it, or have to retype it - rare but for me its way faster to type a longer ~90bit entropy pronounceable password vs a shorter ~90bit fully random password.

I created the list in a simple CVC fashion and filtered for pronounceability using a ~4MB English word list.

[pronounceable_cvc.txt](https://github.com/user-attachments/files/21085092/pronounceable_cvc.txt)

Results in something like that:
----------------
**Fully random examples (~90bit)**

    FP:XMjo@Pqw:MW3
    ;ru3zYAHsh3Hfb4W@


**Pronounceable base (using space for easier readability) example (~90bit)**

    hot cruit smong fluer turn miam teash
    glep kiosk rosp huish yar kion seunt

---

In general I understand why one wouldn't want to add such a feature (that can be misused)! But personally I think it's to restrictive, I need that feature sometimes and as long as one is aware of the implications it comes with.... I like to be able to make choices - and the bits should be a good enough indicator for most people to stay safe....

I also see KeepassXC as the more advanced choice of currently available FOSS PW managers, so it would be a nice fit for it to have such features (ofc thats only my opinion).

But the PW list is a good enough proxy (on a side note: Because of its implications I never used the passphrases function before) 🤗

**KeepassXC is awesome, thanks to all active maintainers in this repo!**

**Comment by kingma-sbw** [comment]
Generate a 2 word passphrase.  And ignore the warning.
Or get Keepass and use this regexp for passwords: `zvcvc[\+\*\%\$\?\&\?\~\-\_\;\:\<\>]zvcv[\+\*\%\$\?\&\?\~\-\_\;\:\<\>]zvcvdd`  typically in 90bit entropy range. Probably more secure than US American missile launch codes

**Comment by s7eph4n** [comment]
> Or get Keepass and use this regexp for passwords: ...

This is terrible advice. The entropy of the password depends on its length and the size of your character range. The non-constant part of your password is 2 characters long, the character range is 14 characters. That results in an entropy of 7.6.
In other words, anybody who wants to brute-force your password would have to check 196 combinations in the worst case. Guess how long any computer produced in the last 50 years or so would need for that. In fact, guess how long it would take me to try all combinations by hand.
The reason Keepass shows an entropy in the 90s is because it assumes a much larger character range.

Edit: In fact, you're using `\?` twice, so the range is really just 13 characters (= 169 combinations, entropy 7.4)
